src/api/layers.api.js

```diff
@@ -135,46 +136,131 @@ export class WMTSLayer extends Layer {
 }
 
 export class WMSLayer extends Layer {
-    constructor(name, id, opacity, baseURL, format) {
-        super(name, LayerTypes.WMS, id, opacity, baseURL);
+    /**
+     * @param {String} name the name of this layer (lang specific)
+     * @param {String} id the ID of this layer in the BOD
+     * @param {Number} opacity the opacity to apply to this layer
+     * @param {String} baseURL the backend to call for tiles
+     * @param {String} format in which image format the backend must be requested
+     * @param {Number} gutter how much of a gutter we want (specific for tiled WMS, if unset this layer will be a single tile WMS)
+     */
+    constructor(name, id, opacity, baseURL, format, gutter = -1) {
+        super(name, LayerTypes.WMS, id, opacity, false, baseURL);
         this.format = format;
+        this.gutter = gutter;
     }
 
     getURL() {
-        return `${this.baseURL}?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&FORMAT=image%2F${this.format}&TRANSPARENT=true&LAYERS=${this.id}&LANG=en`;
+        return `${this.baseURL ? this.baseURL : WMS_BASE_URL}?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&FORMAT=image%2F${this.format}&TRANSPARENT=true&LAYERS=${this.id}&LANG=en`;
     }
 }
 
 export class GeoJsonLayer extends Layer {
-    constructor(name, id, opacity) {
+    constructor(name, id, opacity, geoJsonUrl, styleUrl) {
         super(name, LayerTypes.GEOJSON, id, opacity);
-        this.data = null;
-        this.style = null;
-        this.fetching = false;
+        this.geoJsonUrl = geoJsonUrl;
+        this.styleUrl = styleUrl;
+    }
+
+    getURL() {
+        return this.geoJsonUrl;
     }
 }
 
-const generateClassForLayerConfig = (layerConfig) => {
+export class AggregateSubLayer {
+    /**
+     * @param {String} subLayerId the ID used in the BOD to describe this sub-layer
+     * @param {Layer} layer the sub-layer config (can be a {@link GeoJsonLayer}, a {@link WMTSLayer} or a {@link WMTSLayer})
+     * @param {Number} minResolution in meter/px, at which resolution this sub-layer should start to be visible
+     * @param {Number} maxResolution in meter/px, from which resolution the layer should be hidden
+     */
+    constructor(subLayerId, layer, minResolution = Number.MIN_SAFE_INTEGER, maxResolution = Number.MAX_SAFE_INTEGER) {
```

Comment:
don't you wanna choose realistic values for default min and max resolution (values that are actually in the range of possible resolutions?)

Reply:
I don't know, I wanted to have technically valid values so that I could rely on them being defined.
@ltclm @rebert any hint on that? Is a min res of `0` always going to be smaller than anything defined in the BOD? What about the biggest resolution we have?

Reply:
```
psql -c "SELECT fk_parent_layer, pk_layer, minresolution, maxresolution FROM re3.layers_js WHERE maxresolution 
IS NOT NULL" -d bod_master -h pg-0.dev.bgdi.ch -U pgkogis                                                                              
           fk_parent_layer           |                 pk_layer                  | minresolution | maxresolution 
-------------------------------------+-------------------------------------------+---------------+---------------
 ch.swisstopo-vd.geometa-gemeinde    | ch.swisstopo-vd.geometa-gemeinde_wms      |             0 |            20
 ch.bafu.hydrologie-hintergrundkarte | ch.bafu.hydrologie-hintergrundkarte_wmts2 |             0 |            50
 ch.bafu.hydrologie-hintergrundkarte | ch.bafu.hydrologie-hintergrundkarte_wmts1 |            50 |           750
 ch.swisstopo.swissnames3d           | ch.swisstopo.swissnames3d_wms             |             0 |           1.2
 ch.swisstopo-vd.geometa-grundbuch   | ch.swisstopo-vd.geometa-grundbuch_wms     |             0 |            20
 ch.swisstopo-vd.geometa-nfgeom      | ch.swisstopo-vd.geometa-nfgeom_wms        |             0 |            20
 ch.swisstopo-vd.geometa-standav     | ch.swisstopo-vd.geometa-standav_wms       |             0 |            20
 ch.bfs.gebaeude_wohnungs_register   | ch.bfs.gebaeude_wohnungs_register_wms     |             0 |            10
(8 Zeilen)
```

Reply:
As we are building the new viewer to show the whole world, does it make sense to show those layers past 750 m/px? Or was 750 the "hard limit" of the last viewer and I should continue showing them past this step?

Reply:
so as discussed with Marcel, the limit of max resolution `750` on `ch.bafu.hydrologie-hintergrundkarte` will be removed (it was never reached by `mf-geoadmin3`)